.circleci/config.yml

```diff
@@ -109,68 +189,238 @@ jobs:
             git submodule init
             git submodule update --remote
       - restore_cache:
-          key: 1-gems-{{ checksum "ios/Gemfile.lock" }}-{{ arch }}
+          keys:
+            - 1-gems-{{ arch }}-{{ checksum "ios/Gemfile.lock" }}
+            - 1-gems-{{ arch }}-
       - run:
           command: bundle check || bundle install --path vendor/bundle
           working_directory: ios
       - save_cache:
-          key: 1-gems-{{ checksum "ios/Gemfile.lock" }}-{{ arch }}
+          key: 1-gems-{{ arch }}-{{ checksum "ios/Gemfile.lock" }}
           paths:
             - ios/vendor/bundle
       - restore_cache:
           name: Restore Yarn Package Cache
           keys:
-            - 4-yarn-v1-{{ checksum "yarn.lock" }}-{{ arch }}
+            - 4-yarn-{{ arch }}-{{ checksum "yarn.lock" }}
```

Comment:
Why do we have key `4`here? Does it correspond to `TOTAL_CPUS`?

Reply:
no, if you want to discard a cache you have to change the key in any way. So we just increase some number.